src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs

```diff
@@ -675,8 +675,16 @@ public static String TryGetFullPathToMainApplication()
         /// <param name="moduleBase">Module base address</param>
         public static unsafe String TryGetFullPathToApplicationModule(IntPtr moduleBase)
         {
+#if PLATFORM_UNIX
+            byte* pModuleNameUtf8;
+            int numUtf8Chars = RuntimeImports.RhGetModuleFileName(moduleBase, out pModuleNameUtf8);
+            // Number of UTF-16 chars is always less than or equal to the number of UTF-8 ones,
+            char* pModuleName = stackalloc char[numUtf8Chars];
+            int numChars = System.Text.Encoding.UTF8.GetChars(pModuleNameUtf8, numUtf8Chars, pModuleName, numUtf8Chars);
```

Comment:
This should rather be:

`return System.Text.Encoding.UTF8.GetString((byte*)pModuleNameUtf8, numUtf8Chars);`